commands/utils/docker-endpoint.ts

```diff
@@ -80,22 +81,35 @@ class DockerClient {
         return this.endPoint.getContainer(id);
     }
 
-    public getEngineType(): Thenable<DockerEngineType> {
-        if (process.platform === 'win32') {
-            return new Promise((resolve, reject) => {
-                this.endPoint.info((error, info) => {
-                    if (error) {
-                        return reject(error);
-                    }
-
-                    return resolve(info.OSType === "windows" ? DockerEngineType.Windows : DockerEngineType.Linux);
+    public async getEngineType(): Promise<DockerEngineType> {
+        // tslint:disable-next-line:no-var-self
+        let me = this;
+        let engineType: DockerEngineType;
+        await callWithTelemetryAndErrorHandling('getEngineType', async function (this: IActionContext): Promise<void> {
```

Comment:
What's the reasoning behind a separate telemetry event for this? Couldn't you just include `properties.engineType` in the existing telemetry for the two (as far as I can tell) commands calling this?

Reply:
Somehow seemed like a good idea at the time... :-)